src/LandIce/evaluators/LandIce_w_Resid_Def.hpp

```diff
@@ -47,12 +47,16 @@ namespace LandIce
     TEUCHOS_TEST_FOR_EXCEPTION (dl->side_layouts.find(sideName)==dl->side_layouts.end(), std::runtime_error,
                                 "Error! Basal side data layout not found.\n");
     Teuchos::RCP<Albany::Layouts> dl_side = dl->side_layouts.at(sideName);
+    useCollapsedSidesets = dl_side->useCollapsedSidesets;
 
-    sideBF = decltype(sideBF)(p.get<std::string> ("BF Side Name"), dl_side->node_qp_scalar);
-    side_w_measure = decltype(side_w_measure)(p.get<std::string> ("Weighted Measure Side Name"), dl_side->qp_scalar);
-    side_w_qp  = decltype(side_w_qp)(p.get<std::string> ("w Side QP Variable Name"), dl_side->qp_scalar);
-    basalVerticalVelocitySideQP = decltype(basalVerticalVelocitySideQP)(p.get<std::string>("Basal Vertical Velocity Side QP Variable Name"), dl_side->qp_scalar);
-    normals    = decltype(normals)(p.get<std::string> ("Side Normal Name"), dl_side->qp_vector_spacedim);
+    auto qp_scalar = useCollapsedSidesets ? dl_side->qp_scalar_sideset : dl_side->qp_scalar;
+    auto node_qp_scalar = useCollapsedSidesets ? dl_side->node_qp_scalar_sideset : dl_side->node_qp_scalar;
+    auto qp_vector_spacedim = useCollapsedSidesets ? dl_side->qp_vector_spacedim_sideset : dl_side->qp_vector_spacedim;
+    sideBF = decltype(sideBF)(p.get<std::string> ("BF Side Name"), node_qp_scalar);
```

Comment:
sideBF, side_w_measure, side_w_qp, basalVerticalVelocitySideQP, and normals need to set layout to the _sideset variant if useCollapsedSidesets is set. Looks like it is still using the non-collapsed layout.

Reply:
I think you might have missed the lines right above: to avoid long lines in the construction of the PHX fields, I created the layouts above, with a switch on `useCollapsedSidesets`. Does that make sense?

Reply:
Oh woops, I see that now. I looked at this too quickly and misunderstood. Looks fine as is.